runtime/js/10_permissions.js

```diff
@@ -139,18 +145,26 @@ function cache(desc, state) {
     key += "$";
   }
   if (MapPrototypeHas(statusCache, key)) {
-    const status = MapPrototypeGet(statusCache, key);
-    if (status.state !== state) {
-      status.state = state;
-      status.status.dispatchEvent(new Event("change", { cancelable: false }));
+    const cachedObj = MapPrototypeGet(statusCache, key);
+    if (cachedObj.state !== rawStatus.state) {
+      cachedObj.state = rawStatus.state;
+      cachedObj.status.dispatchEvent(
+        new Event("change", { cancelable: false }),
+      );
+    }
+    if (cachedObj.partial !== rawStatus.partial) {
+      cachedObj.partial = rawStatus.partial;
+      cachedObj.status.dispatchEvent(
+        new Event("change", { cancelable: false }),
+      );
```

Comment:
@nayeemrmn is this correct? Shouldn't there only be a single `change` event emitted if both state and partial change? Or that can't really happen?

Reply:
Good catch, 
```suggestion
    if (cachedObj.state !== rawStatus.state || cachedObj.partial !== rawStatus.partial) {
      cachedObj.state = rawStatus.state;
      cachedObj.partial = rawStatus.partial;
      cachedObj.status.dispatchEvent(
        new Event("change", { cancelable: false }),
      );
```

Reply:
Though yes, partial status can't really change at all since you can't request `--deny`ed scopes, but I don't expect this code to be aware of that